snippets/router-custom-preloading.md

```diff
@@ -0,0 +1,101 @@
+---
+title: Router Custom Preloading
+author: maktarsis
+level: intermediate
+
+tags:
+  - router
+---
+
+# Content
+The topic covers preloading strategy that could help to manipulate modules preloading in the Angular application.
+
+Angular Router provides _PreloadAllModules_ strategy for free, but _NoPreloading_ is enabled by default. This works well, but if our app is very large preloading every module in the background may cause unnecessary data to be loaded. Ideally, we would like to preload the core features on demand. This will allow core features to render immediately when the user navigates to the feature while keeping our core bundle small. As well as we continue to lazy load the rest of the less used features on demand when the user navigates.
+```typescript
+import { Observable, of, timer } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
+
+export class CustomPreloading implements PreloadingStrategy {
+  public preload(route: Route, load: () => Observable<any>): Observable<any> {
+    const loadRoute = delay => delay === false
```

Comment:
This looks really cool, but can we have a more useful example? 
I'm having a hard time understanding what's going on here

Reply:
Feel free to ping me in telegram and chat about how to better simplify it.